api/proto/teleport/access_graph/v1/secrets.proto

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.access_graph.v1;
+
+import "teleport/header/v1/metadata.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/accessgraph/v1;accessgraphv1";
+
+// AuthorizedKey represents a authorized key entry for a certain local user.
+message AuthorizedKey {
+  // metadata is the AuthorizedKey's metadata.
+  teleport.header.v1.Metadata metadata = 1;
+  // kind is a resource kind
+  string kind = 2;
+  // sub_kind is an optional resource sub kind, used in some resources
+  string sub_kind = 3;
+  // version is version
+  string version = 4;
+  // Spec is an AuthorizedKey specification.
+  AuthorizedKeySpec spec = 5;
+}
+
+// AuthorizedKeySpec is the authorized key spec.
+message AuthorizedKeySpec {
+  // host_id is the node identifier and must match the credentials used.
+  string host_id = 1;
+  // key_fingerprint is the SHA256 SSH public key fingerprint.
+  string key_fingerprint = 2;
+  // host_user is the user who can be accessed using the fingerprint above.
+  string host_user = 3;
+}
+
+// PrivateKey represents a private key entry for a certain local user.
+message PrivateKey {
```

Comment:
When is PrivateKey used, in contrast to AuthorizedKey?

Do we really collect the private key itself? It's worth saying that explicitly in the docs.

Reply:
The authorized key is a direct reflection of a node's `/home/$USER/.ssh/authorized_keys` (or its equivalent). It contains details about which user it grants access to, the node ID, and the fingerprint of the public key listed in the file. All servers report their findings back to Teleport.

Conversely, private keys represent a private key but **never** include the key itself. Instead, they contain metadata such as the SHA-256 fingerprint of the SSH public key, the device ID that holds it, and the derivation mode of the SSH public key. The purpose is to report all private keys present on company laptops back to Teleport's access graph and match them against all AuthorizedKeys reported by servers. A match occurs when the PrivateKey fingerprint (the SHA-256 fingerprint of the private's public key) exactly matches the fingerprint in **any** AuthorizedKey.